### PR TITLE
Fix build warnings and errors

### PR DESCRIPTION
**Pull request by gavanderhoorn**
These changes allow me to build the A5 application using the instructions in #7 in a clean Docker container with no errors and only a minimal amount of warnings (in A5 pkgs proper).

Some outstanding issues are with the dependencies of A5.

### CODE REVIEW

File: a5_scan_processing/CMakeLists.txt [hunk 1]
```diff
@@ -13,7 +13,6 @@ find_package(catkin REQUIRED COMPONENTS
 find_package(OpenMP REQUIRED)
 if(OPENMP_FOUND)
   set(OpenMP_FLAGS ${OpenMP_CXX_FLAGS})
-  set(OpenMP_LIBS gomp)
```
Comment (gavanderhoorn) [inline a5_scan_processing/CMakeLists.txt]: According to GCC's OpenMP documentation only the flags (`-fopenmp`) are needed to enable OpenMP for that compiler.

File: a5_utils/CMakeLists.txt [hunk 2]
```diff
@@ -42,10 +42,10 @@ catkin_package(
     tf
     pcl_ros
     moveit_core
-  DEPENDS
-    Eigen3
     a5_msgs
```
Comment (gavanderhoorn) [inline a5_utils/CMakeLists.txt]: `a5_msgs` was first listed as a system dependency, but it's clearly a Catkin pkg.

File: lidar_scanner_node/CMakeLists.txt [hunk 3]
```diff
@@ -11,6 +11,7 @@ find_package(catkin REQUIRED COMPONENTS
   urdf
   interactive_markers
   visualization_msgs
+  pcl_ros
```
Comment (gavanderhoorn) [inline lidar_scanner_node/CMakeLists.txt]: Without this, I got header inclusion errors as headers in `pcl_ros` weren't being found.

Not sure why `VTK` and `PCL` are `find_package(..)`d separately again below. IIUC those should already have been brought in by `pcl_ros`.

Comment (Jmeyer1292) [reply lidar_scanner_node/CMakeLists.txt]: @marip8 Do you have input on this one?

**Comment by gavanderhoorn** [comment]
Most of the remaining build warnings come from `industrial_calibration` (Eigen) and `descartes` (OpenMP and Eigen).

I have a few patches for those as well, but am not sure whether that would be of interest as it's mostly against forks of those repositories (`descartes` fi seem to be mostly fixed upstream already).

**Comment by Jmeyer1292** [comment]
You're welcome to submit patches, but I wouldn't worry about it too much: as you said they're all weird custom branches. We need a general issue to try and reduce the number of these going forward.

File: lidar_scanner_node/CMakeLists.txt (hunk 3 above)
Comment (marip8) [reply lidar_scanner_node/CMakeLists.txt]: Including PCL and VTK separately and not including `pcl_ros` seems like a mistake that I made early on and never corrected. I think the PCL and VTK `find_package()` lines should be removed

Comment (gavanderhoorn) [reply lidar_scanner_node/CMakeLists.txt]: Testing if it builds without those - and links to the correct VTK and PCL.

Comment (gavanderhoorn) [reply lidar_scanner_node/CMakeLists.txt]: Seems to work, I've removed them in 2003af7.

**Comment by gavanderhoorn** [comment]
@Jmeyer1292: made some minor (cosmetic) fixes to build script and manifest of `lidar_scanner_node` in 1f941ef.

I think this can be merged.

File: lidar_scanner_node/CMakeLists.txt [hunk 4]
```diff
@@ -1,49 +1,56 @@
...
+  vtk_viewer
+)
 
 catkin_package(
-  INCLUDE_DIRS include
   CATKIN_DEPENDS
+    interactive_markers
+    pcl_ros
     roscpp
+    roslib
```
Comment (gavanderhoorn) [inline lidar_scanner_node/CMakeLists.txt]: Not sure what `roslib` is used for, but I've left it in the list.

**Comment by Jmeyer1292** [comment]
@gavanderhoorn Thanks for your contributions!

**Pull request merged**